plugins/out_s3/s3.c

```diff
@@ -1066,10 +1108,20 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
     if (ret < 0) {
         /* re-add chunk to list */
         if (chunk) {
-            s3_store_file_unlock(chunk);
             chunk->failures += 1;
+            if (chunk->failures > ctx->ins->retry_limit){
+                less_than_limit = FLB_FALSE;
+            }
+            s3_retry_warn(ctx, tag, chunk->input_name, create_time, less_than_limit);
+            if (less_than_limit == FLB_FALSE) {
+                s3_store_file_unlock(chunk);
+                return FLB_RETRY;
+            }
+            else {
+                s3_store_file_delete(ctx, chunk);
+                return FLB_ERROR;
+            }
         }
```

Comment:
Are you sure this logic is correct?

So if chunk failures is greater than retry limit:
- You retry

If failures is less than retry limit
- You delete the file.

Reply:
Oh good catch... I didn't realize this logic is inverted... @Claych please fix